core/camel-support/src/main/java/org/apache/camel/support/DefaultComponent.java

```diff
@@ -551,6 +551,9 @@ protected boolean useIntrospectionOnEndpoint() {
      * @throws IllegalArgumentException if referenced object was not found in registry.
      */
     public <T> T resolveAndRemoveReferenceParameter(Map<String, Object> parameters, String key, Class<T> type, T defaultValue) {
+        if (parameters.get(key) != null && type.isInstance(parameters.get(key))) {
```

Comment:
Now, when is `getClass` removed, the null check is redundant, as `type.isInstance(null)` returns false. The condition can be simplified to `if (type.isInstance(parameters.get(key)))`